web_app_drawer/static/src/js/web_app_drawer.js

```diff
@@ -0,0 +1,232 @@
+                                'a.oe_menu_leaf, ' +
+                                'a.oe_menu_toggler'
+                                );
+            $clickZones.click($.proxy(this.handleClickZones, this));
+            this.$el.one('drawer.opened', $.proxy(this.onDrawerOpen, this));
+            // Core bus keypress doesn't seem to send arrow keys
+            core.bus.on('keydown', this, this.handleNavKeys);
+        },
+        
+        // It provides handlers to hide drawer when "unfocused"
+        handleClickZones: function() {
+            this.$el.drawer('close');
+            $('.oe_secondary_menus_container')
+                .parent()
+                .collapse('hide');
+        },
+        
+        // It provides keyboard shortcuts for app drawer nav
+        handleNavKeys: function(e) {
+            if (!this.isOpen){
+                return;
+            }
+            if (Object.keys(this.directionCodes).indexOf(e.keyCode.toString()) !== -1) {
+                var $link = this.findAdjacentAppLink(
+                    this.$selectedAppLink,
+                    this.directionCodes[e.keyCode]
+                );
+                this.selectAppLink($link);
+            } else {
+                var buffer = this.handleKeyBuffer(e.keyCode);
+                this.selectAppLink(this.searchAppLinks(buffer));
+            }
+        },
+        
+        // It adds to keybuffer, sets expire timer, and returns buffer
+        handleKeyBuffer: function(keyCode) {
+            this.keyBuffer += String.fromCharCode(keyCode);
+            if (this.keyBufferTimeoutEvent) {
+                clearTimeout(this.keyBufferTimeoutEvent);
+            }
+            this.keyBufferTimeoutEvent = setTimeout(this.clearKeyBuffer, this.keyBufferTime);
+            return this.keyBuffer;
+        },
+        
+        clearKeyBuffer: function() {
+            this.keyBuffer = '';
+        },
+        
+        // It performs close actions & bubbles a ``drawer.closed`` core.bus event
+        onDrawerClose: function() {
+            this.isOpen = false;
+            this.$selectedAppLink = $(false);
+            core.bus.trigger('drawer.closed');
+            this.$el.one('drawer.opened', $.proxy(this.onDrawerOpen, this));
+        },
+        
+        // Generate x and y coords for app icons
+        onDrawerOpen: function() {
+            this.isOpen = true;
+            this.$appLinks = $('.app-drawer-icon-app').parent();
+            this.selectAppLink($(this.$appLinks[0]));
+            this.$appLinks.on('mouseenter', $.proxy(this.selectAppLinkEvent, this));
+            this.$el.one('drawer.closed', $.proxy(this.onDrawerClose, this));
+            core.bus.trigger('drawer.opened');
+        },
+        
+        // It provides a proxy method allowing for app link select w/ events
+        selectAppLinkEvent: function(event) {
+            this.selectAppLink($(event.currentTarget));
```

Comment:
The problem I was running into with selectors was that it seems the core.bus events are not properly handling DOM updates that are performed in code. It feels almost like pre 1.7 without using `delegate`. 

That said, I'll try again with a fresh head just to make sure I wasn't doing something stupid.